text/0000-release-channels.md

```diff
@@ -0,0 +1,244 @@
+prelude, and phase, to import the standard `macro_rules!` macros. In
+the short term this will be worked-around with hacks in the
+compiler. It's likely that these hacks can be removed before 1.0 if
+globs and `macro_rules!` imports become stable.
+
+# Drawbacks
+
+Adding multiple release channels and reducing the release cycle from
+12 to 6 weeks both increase the amount of release engineering work
+required.
+
+The major risk in feature staging is that, at the 1.0 release not
+enough of the language is available to foster a meaningful library
+ecosystem around the stable release. While we might expect many users
+to continue using nightly releases with or without this change, if the
+stable 1.0 release cannot be used in any practical sense it will be
+problematic from a PR perspective. Implementing this RFC will require
+careful attention to the libraries it affects.
+
+Recognizing this risk, we must put in place processes to monitor the
+compatibility of known Cargo crates with the stable release channel,
+using evidence drawn from those crates to prioritize the stabilization
+of features and libraries. [This work has already begun][1], with
+popular feature gates being ungated, and library stabilization work
+being prioritized based on the needs of Cargo crates.
+
+Syntax extensions, lints, and any program using the compiler APIs
+will not be compatible with the stable release channel at 1.0 since it
+is not possible to stabilize `#[plugin_registrar]` in time. Plugins
+are very popular. This pain will partially be alleviated by a proposed
+[Cargo] feature that enables Rust code generation.
+
+[Cargo]: https://github.com/rust-lang/rfcs/pull/403
+[1]: http://blog.rust-lang.org/2014/10/30/Stability.html
+
+# Alternatives
+
+Leave feature gates and experimental APIs exposed to the stable
+channel, as precedented by Haskell, web vendor prefixes, and node.js.
+
+Make the beta channel a compromise between the nightly and stable
+channels, allowing some set of experimental features and APIs. This
+would allow more projects to use a 'more stable' release, but would
+make beta no longer representative of the pending stable release.
+
+# Unresolved questions
+
+The exact method for working around the prelude's use of feature gates
+is undetermined. Fixing [#18102] will complicate the situation as the
+prelude relies on a bug in lint checking to work at all.
+
+[#18102]: https://github.com/rust-lang/rust/issues/18102
+
+Rustdoc disables the feature-staging lints so they don't cause it to
+fail, but I don't know why rustdoc needs to be running lints. It may
+be possible to just stop running lints in rustdoc.
+
+# See Also
+
+* [Stability as a deliverable][1]
+* [Prior work week discussion][2]
+* [Prior detailed description of process changes][3]
+
+[1]: http://blog.rust-lang.org/2014/10/30/Stability.html
+[2]: https://github.com/rust-lang/meeting-minutes/blob/master/workweek-2014-08-18/versioning.md)
```

Comment:
Willing stage airport mata in OUR MY COUNTRY FAVORITE FOR PART CORRECT TIME ANSWER IN BACK GENERAL IN ELECTION.